src/Lab2.ts

```diff
@@ -0,0 +1,418 @@
+import { createMachine, createActor, assign, fromPromise } from "xstate";
+import { speechstate, Settings, Hypothesis } from "speechstate";
+
+const azureCredentials = {
+  endpoint:
+    "https://northeurope.api.cognitive.microsoft.com/sts/v1.0/issuetoken",
+  key: "213d2709cc0b4aa7b6511671d9746b97",
+};
+
+const settings: Settings = {
+  azureCredentials: azureCredentials,
+  asrDefaultCompleteTimeout: 0,
+  locale: "en-US",
+  asrDefaultNoInputTimeout: 10000,
+  ttsDefaultVoice: "en-GB-RyanNeural",
+};
+
+async function fetchFromChatGPT(prompt: string, max_tokens: number) {
+  const myHeaders = new Headers();
+  myHeaders.append(
+    "Authorization",
+    "Bearer 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",
+  );
+  myHeaders.append("Content-Type", "application/json");
+  const raw = JSON.stringify({
+    model: "gpt-3.5-turbo",
+    messages: [
+            {
+              role: "user",
+              content: prompt,
+            },
+          ],
+    temperature: 0,
+    max_tokens: 1000,
+  });
+
+  const response = fetch("https://api.openai.com/v1/chat/completions", {
+    method: "POST",
+    headers: myHeaders,
+    body: raw,
+    redirect: "follow",
+  })
+    .then((response) => response.json())
+    .then((response) => response.choices[0].message.content);
+
+  return response;
+}
+
+interface DMContext {
+  spstRef?: any;
+  lastResult?: Hypothesis[];
+  Destination?: string;
+  StartingPoint?: string;
+  RoutePreference?: string;
+  Stopover?: string;
+  Questions?: string;
+  userInput?: string;
+  recognisedData?: any;
+}
+
+
+
+
+const grammar = {
```

Comment:
not quite sure how this supports over-answering... Can you please provide a test case? The point is that I would like to say that I want to go "from home" "to school with a stopover in Molndal". 

For one-shot request, should I say "from Molndal to Backa avoiding highways and passing through Liseberg"